File: 0169-majority-element/0169-majority-element.cs
public class Solution {
    public int MajorityElement(int[] nums) {
        int n = nums.Length;
        Dictionary<int,int> ans = new Dictionary<int,int>();
        foreach(int x in nums){
            if(!ans.ContainsKey(x)){
                ans[x] = 1;
            }else{
                ans[x]++;
            }
        }

        int result = 0 ;
        foreach(var item in ans){
            if(item.Value > n/2){
                result = item.Key;
            }
        }

        return result;
    }
}

File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int ans = 0 ;
        int n = nums.size();

        unordered_map<int,int> freq ;
        for(int i=0 ; i<nums.size() ; i++){
            freq[nums[i]]++;
        }

        for(int i=0 ; i<n ; i++){
            if( freq[nums[i]] > n/2){
                ans = nums[i];
                break;
            }
        }

        return ans ;
    }
};